lib/reporting/reports/suppliers/helpers/line_items_access_helper.rb

```diff
@@ -25,11 +25,22 @@ def item_order_cycle(line_item)
             line_item.order_cycle
           end
 
+          def suppliers_adjustments(line_item, adjustment_type = 'EnterpriseFee')
+            adjustments = line_item.adjustments
+            return adjustments if adjustment_type == 'Spree::TaxRate'
+
+            supplier_name = supplier(line_item).name
+            adjustments.select do |adjustment|
+              label = adjustment.label
+
+              label.include?('supplier') && label.include?(supplier_name)
+            end
+          end
+
           def adjustments_by_type(line_item, type, included: false)
             total_amount = 0.0
             adjustment_type = type == :tax ? 'Spree::TaxRate' : 'EnterpriseFee'
-            adjustments = line_item.adjustments
-            adjustments.each do |adjustment|
+            suppliers_adjustments(line_item, adjustment_type).each do |adjustment|
```

Comment:
Spree::Adjustment has a bunch of scope ie:
```ruby 
scope :tax, -> { where(originator_type: 'Spree::TaxRate') }
```
Can we refactor this using scope, so we don't have iterate over all adjustment to do some calculation ? ie:
```ruby
adjustments.tax.each do |adjustment|
  # calculation here
end

Reply:
Thanks for the hint. I've conditionally used the scopes in the `suppliers_adjustments` method here:
https://github.com/openfoodfoundation/openfoodnetwork/commit/1c2ac0114e62c46e2d8216dc9f16d04781584375